tests/tests.rs

```diff
@@ -154,9 +155,44 @@ fn run_compile_test(
         }
     };
 
-    output.push(format!("{}.{}", name, ext));
+    let header_suffix = ".header";
+    let header_position = name.rfind(&header_suffix);
+    let is_header = header_position.is_some();
+
+    let source_file = format!(
+        "{}.{}",
+        if is_header {
+            &name[0..header_position.unwrap()]
+        } else {
+            name
+        },
+        &ext
+    );
+    output.push(&source_file);
+
+    if is_header {
```

Comment:
Why are all these changes needed?

Reply:
If they _are_ needed, it may be better to generate the `.c` / `.cpp` file in the tmpdir, rather than checking it in the repo...

Reply:
File with `#pragma once` compile with warning that is treated an error
```
error: #pragma once in main file [-Werror]
    1 | #pragma once
      |         ^~~~
cc1: all warnings being treated as errors
```

Reply:
Ok, instead of these complicated changes, can we just avoid passing `-Werror` for this test?

We can either have a whitelist in the tests.rs file, or some kind of hint in the file name / contents, not sure what's better.

Reply:
(Ideally we'd just pass [`-Wno-pragma-outside-header`](https://clang.llvm.org/docs/DiagnosticsReference.html#wpragma-once-outside-header) and forget about this, but it seems gcc doesn't support this)

Reply:
Ok. I changed this.